File: uniqueNumberOfOccurences.cpp
class Solution
{
    public:
    bool isFrequencyUnique(int n, int arr[])
    {
        /*
        Time: 0.14
        */
        map<long long, int> mp1;
        for(int i=0;i<n;i++){
            mp1[arr[i]]=0;
        }
        for(int i=0;i<n;i++){
            
            mp1[arr[i]]+=1;
        }
        map<long long,int>::iterator iter;
        
        map<int,int> mp2;
        
        for(iter=mp1.begin();iter!=mp1.end();iter++){
        // while(iter!=mp1.end()){
            // cout<<iter->first<<" and "<< iter->second<<endl;
            mp2[iter->second]=0;
        }
        for(iter=mp1.begin();iter!=mp1.end();iter++){
        // while(iter!=mp1.end()){
            // cout<<iter->first<<" and "<< iter->second<<endl;
            mp2[iter->second]+=1;
        }
        
        map<int,int>::iterator iter1;
        // cout<<"Freqs\n";
        for(iter1=mp2.begin();iter1!=mp2.end();iter1++){
        // while(iter!=mp1.end()){
            if(iter1->second>1){
                return false;
            }    
        }
        return true;
    }
};

File: uniqueNumberOfOccurences.py
from typing import List
from collections import Counter as CT

class Solution:
    def isFrequencyUnique(self, n : int, arr : List[int]) -> bool:
        '''
        Both of the ways of implementation work, it's just the way they are written
        Time: 0.09
        '''
        # Implementation 1
        # return len(dict(CT(arr)).values())==len(set(dict(CT(arr)).values()))
        
        # Implementation 2
        d1=dict(CT(arr))
        return len(d1.values())==len(set(d1.values()))